### PR TITLE
LF-4548 Adjust task complete (StepOne) view for movement tasks

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This adds the completion step to Movement Task + (with no further adjustments needed, hopefully) to custom tasks with associated animals. I.e. the ability to change animals on the "Changes needed" view and send the updated animals to the backend should work on any custom task that is passed to the complete view with the property `animalIds`.
- Note there probably is still need for a formatting function to populate the readonly view, similar to what was made for movement task on my previous PR -- I might work on this shortly, but it's not strictly within this ticket's scope

Implemented:
- Readonly and Editable inventories are shown in StepOne if animalIds are present on the task data
- Animal ids are formatted in the complete saga and included in taskData (the object sent without change to the backend for custom tasks)
- Movement task data is additionally formatted using the functions Sayaka already set up for the POST
- Only movement tasks are prohibited from submission with all animals removed (custom tasks should send an empty array and correctly clear all animal associations if the associated animals have all been deselected)

Not implemented:
- Locations are not editable; this was a misunderstanding and the API was never set up to change the task location upon complete; therefore they are umodifiable on Movement Task in the exact same way as every other task

Jira link: https://lite-farm.atlassian.net/browse/LF-4548

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Task/TaskComplete/StepOne.jsx [hunk 1]
```diff
@@ -53,16 +59,25 @@ export default function PureCompleteStepOne({
...
   const taskType = selectedTaskType?.task_translation_key;
 
-  const continueDisabled =
-    taskType === 'SOIL_AMENDMENT_TASK'
-      ? soilAmendmentContinueDisabled(getValues(CHANGES_NEEDED), isValid)
-      : !isValid;
+  const continueDisabled = (() => {
+    switch (taskType) {
+      case 'SOIL_AMENDMENT_TASK':
+        return soilAmendmentContinueDisabled(getValues(CHANGES_NEEDED), isValid);
```
Comment (antsgar) [inline packages/webapp/src/components/Task/TaskComplete/StepOne.jsx:74]:
```suggestion
        return soilAmendmentContinueDisabled(changesRequired, isValid);
```

**Review by antsgar — APPROVED** [review]
Looks and works perfectly!

Really minor thing I'm seeing, the styling in the X icon in the multiselect pills seems off -- maybe I messed something up with my styling suggestion in the previous PR. Anyhow no need to fix here!

**Comment by SayakaOno** [comment]
I'll start reviewing now, could you update the base branch to integration?

**Comment by kathyavini** [comment]
@SayakaOno weird, it should have happened automatically when the base branch was merged!

**Review by SayakaOno — COMMENTED** [review]
Looks great over all!
I'm not sure which is the right ticket for this, but we also need to handle the 0 results on the summary view.

Screenshot 2024-12-04 at 9 55 19 PM

**Comment by SayakaOno** [comment]
I opened a [PR](https://github.com/LiteFarmOrg/LiteFarm/pull/3571) to refetch animals/batches after a movement task is completed to update their location.

If it should be merged to this branch, please let me know!

**Comment by kathyavini** [comment]
Thank you @SayakaOno! I shared that screenshot of the no results block with Loïc and he actually said it was okay! But maybe he didn't closely at the copy 🙂 I agree its weird so I have removed it.

Thank you for the other spots, and for noticing the tag invalidation! I don't thin it necessarily has to be merged together with this branch; integration seems fine!

**Review by antsgar — APPROVED** [review]

**Comment by SayakaOno** [comment]
As I thought, it got conflicts 😂

**Comment by kathyavini** [comment]
Every merge is a conflict now!! 😂🥲

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**